plugins/integration/providers/cielo24/lib/kCielo24FlowManager.php

```diff
@@ -0,0 +1,158 @@
+	{
+		if(in_array($dbBatchJob->getStatus(), array(BatchJob::BATCHJOB_STATUS_FAILED, BatchJob::BATCHJOB_STATUS_DONT_PROCESS, BatchJob::BATCHJOB_STATUS_FINISHED)) && $dbBatchJob->getJobType() == IntegrationPlugin::getBatchJobTypeCoreValue(IntegrationBatchJobType::INTEGRATION))
+		{
+			$providerType = $dbBatchJob->getJobSubType();
+			if ($providerType == Cielo24Plugin::getProviderTypeCoreValue(Cielo24IntegrationProviderType::CIELO24))
+				return true;
+		}
+		return false;
+	}
+	
+	/* (non-PHPdoc)
+	 * @see kBatchJobStatusEventConsumer::updatedJob()
+	 */
+	public function updatedJob(BatchJob $dbBatchJob)
+	{	
+		$data = $dbBatchJob->getData();
+		$providerData = $data->getProviderData();
+		$entryId = $providerData->getEntryId();
+		$partnerId = $dbBatchJob->getPartnerId();
+		$spokenLanguage = $providerData->getSpokenLanguage();
+	
+		$transcript = $this->getAssetsByLanguage($entryId, array(TranscriptPlugin::getAssetTypeCoreValue(TranscriptAssetType::TRANSCRIPT)), $spokenLanguage, true);
+		
+		if($dbBatchJob->getStatus() == BatchJob::BATCHJOB_STATUS_FAILED)
+		{
+			if($transcript)
+			{
+				$transcript->setStatus(AttachmentAsset::FLAVOR_ASSET_STATUS_ERROR);
+				$transcript->save();
+			}
+		}
+		
+		if($dbBatchJob->getStatus() == BatchJob::BATCHJOB_STATUS_DONT_PROCESS)
+		{	
+			if(!$transcript)
+			{
+				$transcript = new TranscriptAsset();
+				$transcript->setType(TranscriptPlugin::getAssetTypeCoreValue(TranscriptAssetType::TRANSCRIPT));
+				$transcript->setEntryId($entryId);
+				$transcript->setPartnerId($partnerId);
+				$transcript->setLanguage($spokenLanguage);
+				$transcript->setContainerFormat(AttachmentType::TEXT);
+			}
+			$transcript->setStatus(AttachmentAsset::ASSET_STATUS_QUEUED);
+			$transcript->save();
+	
+			return true;
+		}
+	
+		$formatsString = $providerData->getCaptionAssetFormats();		
+	
+		if($dbBatchJob->getStatus() == BatchJob::BATCHJOB_STATUS_FINISHED)
+		{
+			$clientHelper = Cielo24Plugin::getClientHelper($providerData->getUsername(), $providerData->getPassword());
+		
+			$remoteJobId = $clientHelper->getRemoteFinishedJobId($entryId);
+			if (!$remoteJobId)
+			{
+				KalturaLog::err('remote content does not exist');
+				return true;     	
+			}
+	
+			$formatsArray = explode(',',$formatsString);
+			$transcriptContent = $clientHelper->getRemoteTranscript($remoteJobId);
+			KalturaLog::debug("transcript content - $transcriptContent");
+			$captionsContentArray = $clientHelper->getRemoteCaptions($remoteJobId, $formatsArray);
+			KalturaLog::debug("captions content - " . print_r($captionsContentArray, true));
+	
+			$captions = $this->getAssetsByLanguage($entryId, array(CaptionPlugin::getAssetTypeCoreValue(CaptionAssetType::CAPTION)), $spokenLanguage);
+		
+			$this->setObjectContent($transcript, $transcriptContent, null, true);
+	
+			foreach ($captionsContentArray as $format => $content)
+			{        
+				$captionFormatConst = constant("KalturaCaptionType::" . $format);
```

Comment:
@rkreich yes, it's a reflection of the providerData's 'CaptionAssetFormats' field which is being validated in the object's constructor.